tests/commands/got.py

```diff
@@ -35,3 +35,32 @@ def test_cmd_got(self):
         res = gdb.execute("got printf", to_string=True)
         self.assertIn("printf", res)
         self.assertNotIn("strcpy", res)
+        self.assertNotIn("/libc", res)
+
+        def checksyms(lines, checked_sections):
+            if not lines:
+                return
+            if "format-string-helper.out" in lines[0]:
+                res = ''.join(lines)
+                self.assertIn(" printf", res)
+                self.assertNotIn(" strcpy", res)
+                checked_sections.append("format-string-helper.out")
+            if "/libc" in lines[0]:
+                res = ''.join(lines)
+                self.assertNotIn(" printf", res)
+                self.assertNotIn(" strcpy", res)
+                checked_sections.append("libc")
+
+        res = gdb.execute("got --all printf", to_string=True)
+        self.assertIn("/libc", res)
+        self.assertIn("format-string-helper.out", res)
+        lines = []
+        checked_sections = []
+        for line in res.splitlines():
```

Comment:
I don't really understand what this is doing, can you please add a comment explaining it?